packages/teleport/src/Nodes/AddNode/useAddNode.ts

```diff
@@ -20,26 +20,28 @@ import useAttempt from 'shared/hooks/useAttemptNext';
 import TeleportContext from 'teleport/teleportContext';
 import { BashCommand } from 'teleport/services/nodes';
 import cfg from 'teleport/config';
-import { JoinToken } from 'teleport/services/joinToken';
+import { JoinToken, Rule } from 'teleport/services/joinToken';
 
 export default function useAddNode(ctx: TeleportContext) {
   const { attempt, run } = useAttempt('processing');
   const isEnterprise = ctx.isEnterprise;
   const version = ctx.storeUser.state.cluster.authVersion;
   const user = ctx.storeUser.state.username;
   const isAuthTypeLocal = !ctx.storeUser.isSso();
-  const [automatic, setAutomatic] = useState(true);
+  const [method, setMethod] = useState<JoinMethod>('iam');
   const [script, setScript] = useState('');
   const [expiry, setExpiry] = useState('');
   const [token, setToken] = useState('');
+  const [iamJoinToken, setIamJoinToken] = useState('');
+  const [iamExpiry, setIamExpiry] = useState('');
```

Comment:
hm, do we really need these two extra states?

Reply:
Hmmm, we don't *need* it. 

Actually, I initially reused the token and expiry that we had already, but I didn't want the users to lose their IAM token if they changed tabs (since going to another tab would trigger a request to a join token with a different method).

I think this is better UX but no strong feelings there, let me know if you think I should change it.